cli/cmd/inject.go

```diff
@@ -281,72 +293,115 @@ func (options *injectOptions) fetchConfigsOrDefault() (*config.All, error) {
 	return api.Config(context.Background(), &public.Empty{})
 }
 
-// overrideConfigs uses command-line overrides to update the provided configs
-func (options *injectOptions) overrideConfigs(configs *config.All) {
+// overrideConfigs uses command-line overrides to update the provided configs.
+// the overrideAnnotations map keeps track of which configs are overridden, by
+// storing the corresponding annotations and values.
+func (options *injectOptions) overrideConfigs(configs *config.All, overrideAnnotations map[string]string) {
```

Comment:
you might as well pull in my fix for #2551 here https://github.com/linkerd/linkerd2/blob/e8db3ce6fefd6b9b079c9bbc2700a67c01513e84/cli/cmd/inject.go#L286-L290

Reply:
Update: we're going to remove the `--linkerd-cni-enabled` inject flag, and instead make it install-only, so you can ignore my comment here.